blockstore.go

```diff
@@ -82,6 +84,16 @@ func gatewayError(err error) error {
 		return fmt.Errorf("%w: %s", gateway.ErrGatewayTimeout, err.Error())
 	}
 
+	// Saturn errors with RetryAfter hint need to be converted to Gateway ones
+	var saturnTooManyRequests *caboose.ErrSaturnTooManyRequests
+	if errors.As(err, &saturnTooManyRequests) {
+		return gateway.NewErrorRetryAfter(saturnTooManyRequests, saturnTooManyRequests.RetryAfter)
+	}
+	var saturnCidCoolDown *caboose.ErrCidCoolDown
+	if errors.As(err, &saturnCidCoolDown) {
+		return gateway.NewErrorRetryAfter(saturnCidCoolDown, saturnCidCoolDown.RetryAfter)
+	}
```

Comment:
:thought_balloon: @hacdias any ideas how we can avoid this mapping?  make caboose return error which is already wrapped in  `ErrorRetryAfter` ?

Reply:
Yes, if caboose already returns `ErrorRetryAfter`, we won't need this.

Reply:
https://github.com/filecoin-saturn/caboose/pull/60